modules/auxiliary/analyze/crack_aix.rb

```diff
@@ -0,0 +1,252 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core/auxiliary/password_cracker'
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Auxiliary::PasswordCracker
+
+  def initialize
+    super(
+      'Name'            => 'Password Cracker: AIX',
+      'Description'     => %Q{
+          This module uses John the Ripper or Hashcat to identify weak passwords that have been
+        acquired from passwd files on AIX systems.  These utilize DES hashing.
+        DES is format 1500 in Hashcat.
+        DES is descrypt in JTR.
+      },
+      'Author'          =>
+        [
+          'theLightCosine',
+          'hdm',
+          'h00die' # hashcat integration
+        ] ,
+      'License'         => MSF_LICENSE,  # JtR itself is GPLv2, but this wrapper is MSF (BSD)
+      'Actions'         =>
+        [
+          ['john', {'Description' => 'Use John the Ripper'}],
+          ['hashcat', {'Description' => 'Use Hashcat'}],
+        ],
+      'DefaultAction' => 'john',
+    )
+
+    register_options(
+      [
+        OptBool.new('INCREMENTAL',[false, 'Run in incremental mode', true]),
+        OptBool.new('WORDLIST',[false, 'Run in wordlist mode', true])
+      ]
+    )
+  end
+
+  def show_command(cracker_instance)
+    if datastore['ShowCommand']
+      if action.name == 'john'
+        cmd = cracker_instance.john_crack_command
+      elsif action.name == 'hashcat'
+        cmd = cracker_instance.hashcat_crack_command
+      end
+      print_status("   Cracking Command: #{cmd.join(' ')}")
+    end
+  end
+
+  def print_results(tbl, cracked_hashes)
+    cracked_hashes.each do |row|
+      unless tbl.rows.include? row
+        tbl << row
+      end
+    end
+    tbl.to_s
+  end
+
+  def run
+    def process_crack(results, hashes, cred, hash_type, method)
+      return results if cred['core_id'].nil? # make sure we have good data
+      # make sure we dont add the same one again
+      add_it = true
+
+      results.each do |r|
+        if r[0] == cred['core_id']
+          add_it = false
+          break
+        end
+      end
+
+      results << [cred['core_id'], hash_type, cred['username'], cred['password'], method] if add_it
```

Comment:
I'm guessing `results` is an `Array` or `Hash`, in which cause you could probably do something like this:

```
unless results.select {|r| r.first == cred['core_id']}.empty?
  results << [cred['core_id'], hash_type, cred['username'], cred['password'], method]
end
```

Rather than:

```
      add_it = true

      results.each do |r|
        if r[0] == cred['core_id']
          add_it = false
          break
        end
      end

      results << [cred['core_id'], hash_type, cred['username'], cred['password'], method] if add_it
```

Untested.

This saves a few lines of code, removes the need for a temporary state variable, and removes the cheeky little `if add_it` EOL conditional.